wayk_proto/src/message/connection_sequence/capabilities.rs

```diff
@@ -322,8 +322,40 @@ impl SystemCapset {
     pub fn new_os_info(os_info: NowSystemOsInfo) -> Self {
         Self {
             flags: SystemCapsetFlags::new_empty().set_os_info(),
-            os_info,
+            os_info: Some(os_info),
+        }
+    }
+}
+
+impl Encode for SystemCapset {
+    fn encoded_len(&self) -> usize {
+        let os_info_len = if let Some(os_info) = &self.os_info {
+            os_info.encoded_len()
+        } else {
+            0
+        };
+
+        mem::size_of::<u32>() + os_info_len
```

Comment:
You can use`self.flags.encoded_len()`, so you don't need to hardcode u32
```suggestion
        flags.encoded_len() + os_info_len
```